src/core/Akka.Streams/Dsl/DelayFlow.cs

```diff
@@ -0,0 +1,190 @@
+using Akka.Streams.Implementation.Fusing;
+using Akka.Streams.Stage;
+
+namespace Akka.Streams.Dsl
+{
+    /// <summary>
+    /// Allows to manage delay and can be stateful to compute delay for any sequence of elements,
+    /// all elements go through <see cref="NextDelay(T)"/> updating state and returning delay for each element
+    /// </summary>
+    /// <typeparam name="T">type of element</typeparam>
+    public interface IDelayStrategy<T>
+    {
+        /// <summary>
+        /// Returns delay for ongoing <paramref name="element"/>, <code>TimeSpan.Zero</code> means passing without delay
+        /// </summary>
+        /// <param name="element">element</param>
+        /// <returns></returns>
+        TimeSpan NextDelay(T element);
+    }
+
+    /// <summary>
+    /// Fixed delay strategy, always returns constant delay for any element.
+    /// </summary>
+    /// <typeparam name="T">type of element</typeparam>
+    public class FixedDelay<T> : IDelayStrategy<T>
+    {
+        private readonly TimeSpan _delay;
+
+        /// <param name="delay">value of the delay</param>
+        public FixedDelay(TimeSpan delay) => _delay = delay;
+
+        /// <inheritdoc/>
+        public TimeSpan NextDelay(T element) => _delay;
+    }
+
+    /// <summary>
+    /// Strategy with linear increasing delay.
+    /// </summary>
+    /// <typeparam name="T">type of element</typeparam>
+    public class LinearIncreasingDelay<T> : IDelayStrategy<T>
+    {
+        private readonly TimeSpan _increaseStep;
+        private readonly Func<T, bool> _needsIncrease;
+        private readonly TimeSpan _initialDelay;
+        private readonly TimeSpan _maxDelay;
+
+        private TimeSpan _delay;
+
+        /// <summary>
+        /// Creates strategy that starts with <paramref name="initialDelay"/> for each element,
+        /// increases by <paramref name="increaseStep"/> every time when <paramref name="needsIncrease"/> returns <code>true</code> up to <paramref name="maxDelay"/>,
+        /// when <paramref name="needsIncrease"/> returns <code>false</code> it resets to <paramref name="initialDelay"/>.
+        /// </summary>
+        /// <param name="increaseStep">step by which delay is increased</param>
+        /// <param name="needsIncrease">if <code>true</code> delay increases, if <code>false</code> delay resets to <paramref name="initialDelay"/></param>
+        /// <param name="initialDelay">initial delay for each of elements</param>
+        /// <param name="maxDelay">limits maximum delay</param>
+        [ApiMayChange]
+        public LinearIncreasingDelay(TimeSpan increaseStep, Func<T, bool> needsIncrease, TimeSpan initialDelay, TimeSpan maxDelay)
+        {
+            if (increaseStep <= TimeSpan.Zero)
+                throw new ArgumentException("Increase step must be positive", nameof(increaseStep));
+
+            if (maxDelay <= initialDelay)
+                throw new Exception($"{nameof(maxDelay)} must be bigger than {nameof(initialDelay)}");
```

Comment:
That‘s not exactly what I meant ;-) 

`throw new ArgumentException("Max delay must be bigger than initial delay", nameof(maxDelay));`